tensortrade/environments/trading_environment.py

```diff
@@ -246,17 +266,26 @@ def reset(self) -> np.array:
 
         return obs
 
-    def render(self, mode='none'):
-        """Renders the environment via matplotlib."""
+    def render(self, episode: int = None, mode: str = 'human'):
+        """Renders the environment as charts or log.
+
+        Arguments:
+            episode: the number of the current episode being rendered (1-based).
+        """
         if mode == 'log':
             self.logger.info('Performance: ' + str(self._portfolio.performance))
-        elif mode == 'chart':
-            if self.viewer is None:
-                raise NotImplementedError()
-
-            self.viewer.render(self.clock.step - 1,
-                               self._portfolio.performance,
-                               self._broker.trades)
+        elif mode == 'human':
+            current_step = self.clock.step - 1
+
+            trades = []
+            for trade in self._broker.trades.values():
```

Comment:
Is there a better way to get these `trades`, without having to re-create the entire list on each time step?

Reply:
I know this is not the best performance and thought of changing it after I finished. However, I thought it's more valuable to get it in the developers' hands fast and left enhancement notes about it in the class comment.

It could be done either by:
- Letting the environment send new trades only to the chart with every render request and the chart adds it to the existing ones or...
- Keep the environment sending the whole trades and let the chart do the diff itself. I'd use pandas `from_dict()` here but that would mean it's better to keep a DataFrame version of the existing trades in the chart to compare to every time.

Either way, for a quick agent training feedback to the community, we could keep the existing approach and enhance later. What do you think?

Reply:
Although the first approach seems lighter on the resources but I prefer the second one because it's more contained. It's more likely for developers to experiment with their custom environments than with the chart component. Keeping tracking the trades difference in the environment rather than the chart is more likely to complicate things for the developers. I'll try to finish it ASAP once we agree on an approach.

Reply:
I made a little enhancement in 2cb9c96 to let the loop skip the existing annotations and add the new ones only. Please check.

Reply:
@notadamking I think I misunderstood your comment. The reason for the following part in `TradingEnvironment`'s `render()`
```
trades = []
for trade in self._broker.trades.values():
    trades.append(trade[0].to_dict())
```
is to let the chart work from a pure `dict` instead of `Trade` classes. I thought it's better to make the chart less reliant on specific framework classes. However, this can easily be changed. Please let me know.

On another note, the chart trade annotation generation now (2cb9c96) skips existing trades to add new ones only. I intend to enhance it by iterating backwards so it returns once an older trade is hit. However, to use bother approaches we need to change `Broker`'s trades from `dict` to `OrderedDict` to ensure sorting. Otherwise I'll need keep a list of integers holding existing trades stes and iterate through the full trades every time to get the new ones. Please let me know what you think.

Reply:
We can change the Broker's trades to an `OrderedDict`, that is fine.

Reply:
As for my original thought here - I was more considered about the performance overhead of converting the entire list of trades into a dict at each time step. Seems like we should just use the original `Trade` object, since these renderers are specific to our `TradingEnvironment`. Thoughts?

Reply:
I had the same concern then too but the tight coupling between the chart and the `Trade` class overweighted it. But I itch when I imagine this conversion repeated with every render. Yes let me change them and update the PR.

Reply:
Dictionary conversion removed in 65bc720. By the way, dictionaries are ordered in Python starting from v3.7. I converted Broker trades to OrderedDict though because the version used here is >= 3.6.